platform/core/src/studies/services/wado/retrieveMetadataLoaderSync.js

```diff
@@ -0,0 +1,78 @@
+import { api } from 'dicomweb-client';
+import DICOMWeb from '../../../DICOMWeb/';
+import * as StudyUtils from '../../studyUtils';
+import RetrieveMetadataLoader from './retrieveMetadataLoader';
+
+/**
+ * Class for sync load of study metadata.
+ * It inherits from RetrieveMetadataLoader
+ *
+ * A list of loaders (getLoaders) can be created so, it will be applied a fallback load strategy.
+ * I.e Retrieve metadata using all loaders possibilities.
+ */
+export default class RetrieveMetadataLoaderSync extends RetrieveMetadataLoader {
+  getOptions() {
+    const { studyInstanceUID, filters } = this;
+
+    const options = {
+      studyInstanceUID,
+    };
+
+    const { seriesInstanceUID } = filters;
+    if (seriesInstanceUID) {
+      options['seriesInstanceUID'] = seriesInstanceUID;
+    }
+
+    return options;
+  }
+
+  /**
+   * @returns {Array} Array of loaders. To be consumed as queue
+   */
+  *getLoaders() {
+    const loaders = [];
+    const { filters: { seriesInstanceUID } = {}, client } = this;
+
+    if (seriesInstanceUID) {
+      loaders.push(client.retrieveSeriesMetadata);
+    }
+
+    loaders.push(client.retrieveStudyMetadata);
```

Comment:
I think I see where/how we're ultimately filtering our result set in the `async` implementation. I'm pretty sure this is how/where we accomplish this here. I'm not 100% sure I understand why it works here. Can you give me a high level overview?

Reply:
yes.. lets do it..